engine/src/flutter/shell/platform/embedder/embedder_semantics_update.cc

```diff
@@ -238,6 +278,8 @@ void EmbedderSemanticsUpdate2::AddNode(const SemanticsNode& node) {
       CreateStringAttributes(node.increasedValueAttributes);
   auto decreased_value_attributes =
       CreateStringAttributes(node.decreasedValueAttributes);
+  FlutterSemanticsFlags flags = ConvertToFlutterSemanticsFlags(node.flags);
+  flags_.push_back(flags);
```

Comment:
In theory, I _think_ you can do this to construct in-place in the vector, or at least take advantage of RVO in C++:
```cc
flags_.emplace_back(ConvertToFlutterSemanticsFlags(node.flags));
```

See below, since I have the same concern as @loic-sharma about returning a raw pointer into the flags array. If we go with the `unique_ptr` approach, this may be equally useful.